### PR TITLE
vNIC tests enhancement

### PR DESCRIPTION
**Pull request by abdhaleegit**
[1/2] Get HMC ip and server and dlpar package fix
[2/2] vNIC: add dlpar add/remove of vNIC virtual slot test

### CODE REVIEW

**Comment by abdhaleegit** [comment]
File /etc/avocado/sysinfo/commands does not exist.
File /etc/avocado/sysinfo/files does not exist.
File /etc/avocado/sysinfo/profilers does not exist.
INIT 7-network_virtualization.py:NetworkVirtualization.test_vnic_dlpar;run-9d4a
PARAMS (key=timeout, path=*, default=None) => None
Test metadata:
  filename: /root/avocado-misc-tests/io/net/virt-net/network_virtualization.py
  teststmpdir: /var/tmp/avocado_NbZZNv
  workdir: /var/tmp/avocado_hWdUIl/avocado_job_zEkxRi/7-network_virtualization.py_NetworkVirtualization.test_vnic_dlpar_run-9d4a
START 7-network_virtualization.py:NetworkVirtualization.test_vnic_dlpar;run-9d4a
DATA (filename=output.expected) => NOT FOUND (data sources: variant, test, file)
Test is running on: rhel
yum version: 3.4.3
Running 'rpm -q ksh'
[stdout] ksh-20120801-139.el7.ppc64le
Command 'rpm -q ksh' finished with 0 after 0.010381937027s
Running 'rpm -q src'
[stdout] src-3.2.4.3-19120.ppc64le
Command 'rpm -q src' finished with 0 after 0.00980806350708s
Running 'rpm -q rsct.basic'
[stdout] rsct.basic-3.2.4.3-19120.ppc64le
Command 'rpm -q rsct.basic' finished with 0 after 0.0101661682129s
Running 'rpm -q rsct.core.utils'
[stdout] rsct.core.utils-3.2.4.3-19120.ppc64le
Command 'rpm -q rsct.core.utils' finished with 0 after 0.0103089809418s
Running 'rpm -q rsct.core'
[stdout] rsct.core-3.2.4.3-19120.ppc64le
Command 'rpm -q rsct.core' finished with 0 after 0.0124158859253s
Running 'rpm -q DynamicRM'
[stdout] DynamicRM-2.0.7-2.ppc64le
Command 'rpm -q DynamicRM' finished with 0 after 0.00999283790588s
Running 'rpm -q powerpc-utils'
[stdout] powerpc-utils-1.3.4-8.el7.ppc64le
Command 'rpm -q powerpc-utils' finished with 0 after 0.0105841159821s
Running 'lsrsrc IBM.MCP HMCIPAddr'
[stdout] Resource Persistent Attributes for IBM.MCP
[stdout] resource 1:
[stdout] 	HMCIPAddr = "9.3.90.231"
Command 'lsrsrc IBM.MCP HMCIPAddr' finished with 0 after 0.0842459201813s
PARAMS (key=hmc_pwd, path=*, default=None) => 'abc123'
PARAMS (key=hmc_username, path=*, default=None) => 'hscroot'
PARAMS (key=lpar, path=*, default=None) => 'ltczep5-lp7-R77-Abdul'
Running the command on hmc lssyscfg -r sys  -F name
PARAMS (key=slot_num, path=*, default=None) => 10
PARAMS (key=vios_names, path=*, default=None) => 'ltczep5-vios1,ltczep5-vios2'
PARAMS (key=sriov_ports, path=*, default=None) => '0,0'
PARAMS (key=sriov_adapters, path=*, default=None) => 'U78D4.ND1.CSS10M1-P1-C3,U78D4.ND1.CSS10M1-P1-C5'
PARAMS (key=bandwidth, path=*, default=None) => 2
PARAMS (key=vnic_test_count, path=*, default=1) => 1
PARAMS (key=num_of_dlpar, path=*, default=1) => 2
PARAMS (key=device_ip, path=*, default=None) => '179.1.1.79'
PARAMS (key=mac_id, path=*, default=02:03:03:03:03:01) => '02:03:03:04:00:04'
PARAMS (key=netmask, path=*, default=None) => '255.255.255.0'
PARAMS (key=peer_ip, path=*, default=None) => '179.1.1.116'
Running the command on hmc uname -a
Running the command on hmc lssyscfg -m ltczep5 -r lpar --filter lpar_names=ltczep5-lp7-R77-Abdul -F lpar_id
Running the command on hmc lssyscfg -m ltczep5 -r lpar --filter lpar_names=ltczep5-vios1 -F lpar_id
Running the command on hmc lssyscfg -m ltczep5 -r lpar --filter lpar_names=ltczep5-vios2 -F lpar_id
Running the command on hmc lshwres -m ltczep5 -r sriov --rsubtype adapter -F               phys_loc:adapter_id
Running 'startsrc -g rsct'
[stdout] 0513-029 The ctcas Subsystem is already active.
Command 'startsrc -g rsct' finished with 0 after 3.00222682953s
[stdout] Multiple instances are not supported.
[stdout] 0513-029 The ctrmc Subsystem is already active.
[stdout] Multiple instances are not supported.
Running 'startsrc -g rsct_rm'
[stdout] 0513-029 The IBM.MgmtDomainRM Subsystem is already active.
Command 'startsrc -g rsct_rm' finished with 0 after 3.00828313828s
[stdout] Multiple instances are not supported.
[stdout] 0513-059 The IBM.ERRM Subsystem has been started. Subsystem PID is 8785.
[stdout] 0513-059 The IBM.AuditRM Subsystem has been started. Subsystem PID is 8786.
[stdout] 0513-029 The IBM.HostRM Subsystem is already active.
[stdout] Multiple instances are not supported.
[stdout] 0513-059 The IBM.SensorRM Subsystem has been started. Subsystem PID is 8787.
[stdout] 0513-029 The IBM.ServiceRM Subsystem is already active.
[stdout] Multiple instances are not supported.
[stdout] 0513-029 The IBM.DRM Subsystem is already active.
[stdout] Multiple instances are not supported.
[stdout] 0513-029 The IBM.ConfigRM Subsystem is already active.
[stdout] Multiple instances are not supported.
Running 'lssrc -a'
[stdout] Subsystem         Group            PID     Status 
Command 'lssrc -a' finished with 0 after 0.00312399864197s
[stdout]  ctrmc            rsct             10405   active
[stdout]  IBM.HostRM       rsct_rm          10936   active
[stdout]  IBM.DRM          rsct_rm          11013   active
[stdout]  IBM.ServiceRM    rsct_rm          11036   active
[stdout]  IBM.MgmtDomainRM rsct_rm          11040   active
[stdout]  ctcas            rsct             40144   active
[stdout]  IBM.ConfigRM     rsct_rm          40174   active
[stdout]  IBM.ERRM         rsct_rm          8785    active
[stdout]  IBM.AuditRM      rsct_rm          8786    active
[stdout]  IBM.SensorRM     rsct_rm          8787    active
Running the command on hmc lshwres -r virtualio -m ltczep5 --rsubtype vnic --level lpar                --filter lpar_names=ltczep5-lp7-R77-Abdul -F slot_num,backing_device_states
Running the command on hmc lshwres -r virtualio -m ltczep5 --rsubtype vnic --level lpar                --filter lpar_names=ltczep5-lp7-R77-Abdul -F backing_devices
Running 'ls -l /sys/class/net/ |                                            grep eth1 | cut -d '/' -f                                            5'
[stdout] 3000000a
Command 'ls -l /sys/class/net/ |                                            grep eth1 | cut -d '/' -f                                            5' finished with 0 after 0.00348806381226s
Running 'lsslot'
[stdout] # Slot                      Description       Linux Name    Device(s)
Command 'lsslot' finished with 0 after 0.073558807373s
[stdout] U9040.MR9.133CF0X-V7-C0     Virtual I/O Slot  30000000      vty
[stdout] U9040.MR9.133CF0X-V7-C2     Virtual I/O Slot  30000002      l-lan
[stdout] U9040.MR9.133CF0X-V7-C10    Virtual I/O Slot  3000000a      vnic
[stdout] U9040.MR9.133CF0X-V7-C1207  Virtual I/O Slot  300004b7      vfc-client
[stdout] U9040.MR9.133CF0X-V7-C1307  Virtual I/O Slot  3000051b      vfc-client
[stdout] U9040.MR9.133CF0X-V7-C1407  Virtual I/O Slot  3000057f      vfc-client
[stdout] U9040.MR9.133CF0X-V7-C2207  Virtual I/O Slot  3000089f      vfc-client
[stdout] U9040.MR9.133CF0X-V7-C2307  Virtual I/O Slot  30000903      vfc-client
[stdout] U9040.MR9.133CF0X-V7-C2407  Virtual I/O Slot  30000967      vfc-client
Running 'drmgr -r -c slot -s U9040.MR9.133CF0X-V7-C10 -w 5 -d 1'
[stderr] Validating I/O DLPAR capability...yes.
[stderr] failed to open /sys/bus/pci/slots/U9040.MR9.133CF0X-V7-C10/power: No such file or directory
[stderr] failed to disable hotplug children
Command 'drmgr -r -c slot -s U9040.MR9.133CF0X-V7-C10 -w 5 -d 1' finished with 0 after 1.55344820023s
Running 'drmgr -a -c slot -s U9040.MR9.133CF0X-V7-C10 -w 5 -d 1'
[stderr] Validating I/O DLPAR capability...yes.
Command 'drmgr -a -c slot -s U9040.MR9.133CF0X-V7-C10 -w 5 -d 1' finished with 0 after 0.248851060867s
Running 'drmgr -r -c slot -s U9040.MR9.133CF0X-V7-C10 -w 5 -d 1'
[stderr] Validating I/O DLPAR capability...yes.
[stderr] failed to open /sys/bus/pci/slots/U9040.MR9.133CF0X-V7-C10/power: No such file or directory
[stderr] failed to disable hotplug children
Command 'drmgr -r -c slot -s U9040.MR9.133CF0X-V7-C10 -w 5 -d 1' finished with 0 after 1.51187109947s
ORunning 'drmgr -a -c slot -s U9040.MR9.133CF0X-V7-C10 -w 5 -d 1'
[stderr] Validating I/O DLPAR capability...yes.
Command 'drmgr -a -c slot -s U9040.MR9.133CF0X-V7-C10 -w 5 -d 1' finished with 0 after 0.247678041458s
Running 'ip addr add 179.1.1.79/255.255.255.0 dev eth1;ip link set eth1 up'
Command 'ip addr add 179.1.1.79/255.255.255.0 dev eth1;ip link set eth1 up' finished with 0 after 0.00345706939697s
Network virtualized device eth1 is up
Successfully configured the Network                               Virtualized device
Running 'ping -I eth1 179.1.1.116 -c 5'
[stdout] PING 179.1.1.116 (179.1.1.116) from 179.1.1.79 eth1: 56(84) bytes of data.
[stdout] 64 bytes from 179.1.1.116: icmp_seq=1 ttl=64 time=0.276 ms
[stdout] 64 bytes from 179.1.1.116: icmp_seq=2 ttl=64 time=0.097 ms
[stdout] 64 bytes from 179.1.1.116: icmp_seq=3 ttl=64 time=0.121 ms
[stdout] 64 bytes from 179.1.1.116: icmp_seq=4 ttl=64 time=0.120 ms
[stdout] 64 bytes from 179.1.1.116: icmp_seq=5 ttl=64 time=0.106 ms
Command 'ping -I eth1 179.1.1.116 -c 5' finished with 0 after 3.99953508377s
[stdout] 
[stdout] --- 179.1.1.116 ping statistics ---
[stdout] 5 packets transmitted, 5 received, 0% packet loss, time 3996ms
[stdout] rtt min/avg/max/mdev = 0.097/0.144/0.276/0.066 ms
DATA (filename=output.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
PASS 7-network_virtualization.py:NetworkVirtualization.test_vnic_dlpar;run-9d4a

**Comment by abdhaleegit** [comment]
[dlparrunlogs.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/3285117/dlparrunlogs.txt)

File: io/net/virt-net/network_virtualization.py [hunk 1]
```diff
@@ -545,6 +569,16 @@ def find_device_id(self):
...
 
+    def find_virtual_slot(self, dev_id):
+        """
+        finds the virtual slot for the given virtual ID
+        """
+        output = process.system_output("lsslot", ignore_status=True,
+                                       shell=True, sudo=True)
+        for slot in output.split('\n'):
+            if dev_id in slot:
+                return slot.split(' ')[0]
```
Comment (narasimhan-v) [inline io/net/virt-net/network_virtualization.py]: Please handle the case when this slot is not got.
Either here or the function call.

File: io/net/virt-net/network_virtualization.py [hunk 2]
```diff
@@ -350,6 +351,21 @@ def test_backingdevremove(self):
...
+    def test_vnic_dlpar(self):
+        '''
+        Perform vNIC device hot add and hot remove using drmgr command
+        '''
+        self.update_backing_devices()
+        dev_id = self.find_device_id()
+        slot = self.find_virtual_slot(dev_id)
+        for _ in range(self.num_of_dlpar):
+            self.drmgr_vnic_dlpar('-r', slot)
+            time.sleep(5)
```
Comment (narasimhan-v) [inline io/net/virt-net/network_virtualization.py]: can we find an alternative for this sleep ?

Comment (abdhaleegit) [reply io/net/virt-net/network_virtualization.py]: How about this ?
wait.wait_for(self.drmgr_vnic_dlpar('-r', slot))

Comment (narasimhan-v) [reply io/net/virt-net/network_virtualization.py]: Does that work ?

**Comment by abdhaleegit** [comment]
@narasimhan-v changes accommodated.

**Comment by narasimhan-v** [comment]
@abdhaleegit Travis Failure

**Comment by narasimhan-v** [comment]
@abdhaleegit please add job.log, and we can merge this.

**Pull request merged**